eng/packages/http-client-csharp/generator/TestProjects/Local/Basic-TypeSpec/src/Generated/BasicTypeSpecClientListWithContinuationTokenHeaderResponseAsyncCollectionResultOfT.cs

```diff
@@ -0,0 +1,86 @@
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace BasicTypeSpec
+{
+    internal partial class BasicTypeSpecClientListWithContinuationTokenHeaderResponseAsyncCollectionResultOfT : AsyncPageable<ThingModel>
+    {
+        private readonly BasicTypeSpecClient _client;
+        private readonly string _token;
+        private readonly RequestContext _context;
+
+        /// <summary> Initializes a new instance of BasicTypeSpecClientListWithContinuationTokenHeaderResponseAsyncCollectionResultOfT, which is used to iterate over the pages of a collection. </summary>
+        /// <param name="client"> The BasicTypeSpecClient client used to send requests. </param>
+        /// <param name="token"></param>
+        /// <param name="context"> The request options, which can override default behaviors of the client pipeline on a per-call basis. </param>
+        public BasicTypeSpecClientListWithContinuationTokenHeaderResponseAsyncCollectionResultOfT(BasicTypeSpecClient client, string token, RequestContext context)
+        {
+            _client = client;
+            _token = token;
+            _context = context;
+        }
+
+        /// <summary> Gets the pages of BasicTypeSpecClientListWithContinuationTokenHeaderResponseAsyncCollectionResultOfT as an enumerable collection. </summary>
+        /// <param name="continuationToken"> The continuation token. </param>
+        /// <param name="pageSizeHint"> The page size hint. </param>
+        /// <returns> The pages of BasicTypeSpecClientListWithContinuationTokenHeaderResponseAsyncCollectionResultOfT as an enumerable collection. </returns>
+        public override async IAsyncEnumerable<Page<ThingModel>> AsPages(string continuationToken, int? pageSizeHint)
+        {
+            string nextLink = continuationToken;
+            do
+            {
+                Response response = await GetNextResponse(pageSizeHint, nextLink).ConfigureAwait(false);
+                if (response is null)
+                {
+                    yield break;
+                }
+                ListWithContinuationTokenHeaderResponseResponse items = (ListWithContinuationTokenHeaderResponseResponse)response;
+                nextLink = (response.Headers.TryGetValue("next-token", out string value) ? value : null).ToString();
+                yield return Page<ThingModel>.FromValues((IReadOnlyList<ThingModel>)items.Things, nextLink, response);
+            }
+            while (!string.IsNullOrEmpty(nextLink));
+        }
+
+        /// <summary> Get response from next link. </summary>
+        /// <param name="pageSizeHint"> The page size hint. </param>
+        /// <param name="continuationToken"> The continuation token. </param>
+        private async ValueTask<Response> GetNextResponse(int? pageSizeHint, string continuationToken)
+        {
+            HttpMessage message = _client.CreateListWithContinuationTokenHeaderResponseRequest(_token, _context);
```

Comment:
Shouldn't this be passing the continuationToken?

Reply:
_token is the continuation token

Reply:
_token is the initial token. `continuationToken` is the current token.